### ISSUE TITLE
Regression when resolving the module ids of externalised dynamic imports of NPM packages in rollup@0.66.2

### ISSUE THREAD

**Issue by sebinsua** [rollup/rollup#2481]
- Rollup Version: `0.66.2`
- Operating System (or Browser): OSX
- Node Version: `10.10.0`

There seems to have been a regression between `0.64.1` and `0.66.2` in the output of externalised dynamic imports of NPM packages when also using `rollup-plugin-node-resolve`.

Previously neither static imports or dynamic imports would call `rollup-plugin-node-resolve#resolveId` on an externalised module. However, since updating `rollup` I've noticed that external dynamic imports are now inconsistent with static imports and do resolve their `module.id` into an absolute path.

This causes the production of JavaScript which can only be installed on the machine they were built on.

The bug might have been introduced within [the Plugin System Refactoring](https://github.com/rollup/rollup/pull/2382) by @guybedford or in subsequent changes.

In the short-term I am able to explicitly cause `rollup-plugin-node-resolve` to treat certain packages as being externalised by passing in a regex of `/^(?!externalised-pkg-name).*$/` into `options#only`.

### How Do We Reproduce?

Clone [`sebinsua/rollup-resolve-externalised-dynamic-imports`](https://github.com/sebinsua/rollup-resolve-externalised-dynamic-imports), run `npm install` and then run `npm run build` and look at the generated JavaScript.

### Expected Behavior

```js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));

var test = function test() {
  return React;
};

var someDynamicImport = function someDynamicImport() {
  var test = Promise.resolve(require('react'));
  return test;
};

exports.test = test;
exports.someDynamicImport = someDynamicImport;
```

### Actual Behavior

```js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));

var test = function test() {
  return React;
};

var someDynamicImport = function someDynamicImport() {
  var test = Promise.resolve({ default: require("/Users/my-username/dev/rollup-resolve-externalised-dynamic-imports/node_modules/react/index.js") });
  return test;
};

exports.test = test;
exports.someDynamicImport = someDynamicImport;
```

**Comment by sebinsua** [comment]
I ['fixed' it here](https://github.com/sebinsua/rollup/tree/dynamic-import-chunking-not-resolving).

I'm not sure whether it's right though? I don't fully understand the data structures I updated. I was just following my intuition that dynamic imports which return `true` for `this.isExternal` should store a `dynamicImportExpression` gainst their `ExternalModule` instead of a `replacement`. However, this caused another test to have to change, so maybe this is wrong...

**Comment by lukastaegert** [comment]
Thanks for finding this issue and for your digging around! As it turns out, the cause of the issue was that after the plugin system refactoring, external ids were no longer recognized as such but passed to the plugins for further resolution. Fix at #2505.

**Comment by sebinsua** [comment]
Thanks a lot for the fix. Looking forwards to the new version. :)
